tuf/api/metadata.py

```diff
@@ -116,6 +116,33 @@ def __init__(self, signed: T, signatures: Dict[str, Signature]):
         self.signed: T = signed
         self.signatures = signatures
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Metadata):
+            return False
+
+        # Verifies that the signatures for both objects are the same.
+        # TODO: Use Signature.__eq__ (secure-systems-lab/securesystemslib#383)
+        # pylint: disable=unidiomatic-typecheck
+        if not (type(self.signatures) == type(other.signatures) == dict):
```

Comment:
I guess the idiomatic check would be

```suggestion
        if not (isinstance(self.signatures, dict) and isinstance(other.signatures, dict)):
```

Reply:
Yes, but it becomes too long more than 80 characters... When I format it with black it looks like this:

```python
        if not (
            isinstance(self.signatures, dict)
            and isinstance(other.signatures, dict)
        ):
```
which is ugly.
That's why I decided to use `type` as it's shorter and tripple `==` even though it's not idiomatic, but it does the same job.

Reply:
I do have a slight preference because you use `isinstance`-checks everywhere else in your `__eq__` implementations, so it seems consistent to also use it here. That said, I don't think that the subtle difference between `isinstance` and `== type` in this case will ever be a problem. So you decide.

Reply:
Will follow your suggestion. :))